Plans: Add advertising for live courses throughout Business signup

jeremeylduvall opened the pull request:
**Note:** This is dependent on the addition of the new /courses page ([milestone](https://github.com/Automattic/wp-calypso/milestone/121)). Let's wait to merge until the /courses page is up and running.

The new live courses at `wordpress.com/help/courses` will only be available to Business customers. This PR adds mention of live courses at several places throughout the Business signup so that Business customers are aware of the feature and how to take advantage. Discussion in p3fqKv-3uP-p2.

## To test

1. Checkout this branch
2. Upgrade a site to Business
3. Look at the three screens mentioned below to make sure the copy is appearing correctly.
4. While running the branch, also take a look at a site with a Personal or Premium bundle. You shouldn't see the live courses feature mentioned below.

Specifically, it adds mention of the feature in three places:

**On the /my-plan/ page**
Example: `http://calypso.localhost:3000/plans/my-plan/site.wordpress.com`

![screen shot 2016-08-31 at 2 58 30 pm](https://cloud.githubusercontent.com/assets/7240478/18146140/66fbd846-6f8b-11e6-9d5a-2207027f0c6c.png)

**As a feature on /plans**
Example: `http://calypso.localhost:3000/plans/site.wordpress.com`

![screen shot 2016-08-31 at 2 59 15 pm](https://cloud.githubusercontent.com/assets/7240478/18146172/822395fa-6f8b-11e6-9124-013b1d1681a5.png)

**On the Thank You page after checkout**
Example: `http://calypso.localhost:3000/checkout/thank-you/site.wordpress.com/555555`

![screen shot 2016-08-31 at 2 59 45 pm](https://cloud.githubusercontent.com/assets/7240478/18146199/a942ffea-6f8b-11e6-8d14-8b5140d09723.png)


Test live: https://calypso.live/?branch=add/courses-page-business-signup

jeremeylduvall commented:
Just a note that I fixed the typo in the thank you page above (`Happiness Engineersto`).

dllh commented:
@omarjackman @jordwest could one of you give this a look?

jordwest commented on client/blocks/product-purchase-features/product-purchase-features-list/index.jsx:
```diff
@@ -77,6 +78,10 @@ export default class ProductPurchaseFeaturesList extends Component {
 				key="customizeThemeFeature"
 			/>,
+			<LiveCourses
+				isBusinessPlan
```
Is this prop necessary? It seems to be used in `AdvertisingRemoved` to alter the feature description for Premium and Business, but doesn't look like it's used inside the `LiveCourses` component

jordwest commented on client/lib/plans/constants.js:
```diff
@@ -421,6 +424,16 @@ export const featuresList = {
+		getSlug: () => FEATURE_LIVE_COURSES,
+		getTitle: () => i18n.translate( 'Attend live courses' ),
+		getDescription: () => i18n.translate(
+			"Attend live courses led by Happiness Engineers to get the most out of your site."
```
There's a lint error here - we should use single-quotes to define strings unless there's a single-quote contained in the string

jordwest commented on client/lib/plans/constants.js:
```diff
@@ -421,6 +424,16 @@ export const featuresList = {
+		getDescription: () => i18n.translate(
+			"Attend live courses led by Happiness Engineers to get the most out of your site."
+		),
+		getStoreSlug: () => 'live_courses',
```
I was curious what `getStoreSlug` is used for, but couldn't find it being called anywhere in Calypso. A full text search returns only this file. Maybe we can leave it out until it's needed?

jordwest commented:
Gave this a spin and everything worked as expected, looks great 👍 
Left a few minor comments, but otherwise it's a 🚢  from me when the time comes.

omarjackman commented on client/my-sites/upgrades/checkout-thank-you/business-plan-details.jsx:
```diff
@@ -22,6 +23,15 @@ const BusinessPlanDetails = ( { selectedSite, sitePlans, selectedFeature } ) =>
+					'to get the most out of your site.'
+				) }
+				buttonText={ i18n.translate( 'Register for a course' ) }
+				href={ support.CALYPSO_COURSES } />
```
I wonder if we should add any tracks pings here. I didn't see any for the other purchase details but maybe since we're tracking course clicks from /help we can do it here also.

jeremeylduvall commented:
Addressed the comments above and added Tracks pings per @omarjackman's comment.

omarjackman commented on client/blocks/product-purchase-features/product-purchase-features-list/live-courses.jsx:
```diff
@@ -7,9 +7,14 @@ import { localize } from 'i18n-calypso';
 import support from 'lib/url/support';
 
+function trackCoursesButtonClick() {
+	analytics.tracks.recordEvent( 'calypso_my_plan_courses_click' );
```
I'd expect something like `calypso_purchase_features_courses_button_click`. Sounds kind of long but I only bring it up because I'm not sure where the `my_plan` part comes from. I'm sure there's something better than what I'm suggesting if what you have isn't already. The name here could get tricky because I'm sure at some point if we start tracking clicks for other purchase feature list items we'll want to group them together in a nice way.

omarjackman commented:
I left a comment about the tracks ping name `calypso_my_plan_courses_click` but Its not a big deal so you get a :ship: from me too if you feel the name is descriptive enough.

jeremeylduvall commented:
I used `my_plan` initially because of the URL scheme (https://wordpress.com/plans/my-plan/site.wordpress.com). However, you raise a good point that `my_plan` isn't super intuitive and we will probably want to have a repeatable format. I switched it to `calypso_plan_features_courses_click`. We can use the same format (`calypso_plan_features_google_analytics_click`) for other items on that page moving forward if we want to add them.

Rebased and squashed a bit. Will wait for checks to run and then merge [since /help/courses is live.](https://github.com/Automattic/wp-calypso/commit/d55d434a52aa2db095ce3c50b0129ae9b3360fa2)
